src/openvino-dot/openvino-dot-metadata.js

```diff
@@ -0,0 +1,189 @@
+        if (data) {
+            var items = JSON.parse(data);
+            if (items) {
+                items.forEach((item) => {
+                    if (item.name && item.schema) {
+                        var name = item.name;
+                        var schema = item.schema;
+                        this._map[name] = schema;
+                    }
+                });
+            }
+        }
+    }
+
+    getOperatorCategory(operator) {
+        var schema = this._map[operator];
+        if (schema && schema.category) {
+            return schema.category;
+        }
+        return null;
+    }
+
+    getOperatorDocumentation(operator) {
+        var schema = this._map[operator];
+        if (schema) {
+            schema = JSON.parse(JSON.stringify(schema));
+            schema.name = operator;
+            if (schema.description) {
+                schema.description = marked(schema.description);
+            }
+            if (schema.attributes) {
+                schema.attributes.forEach((attribute) => {
+                    if (attribute.description) {
+                        attribute.description = marked(attribute.description);
+                    }
+                });
+            }
+            if (schema.inputs) {
+                schema.inputs.forEach((input) => {
+                    if (input.description) {
+                        input.description = marked(input.description);
+                    }
+                });
+            }
+            if (schema.outputs) {
+                schema.outputs.forEach((output) => {
+                    if (output.description) {
+                        output.description = marked(output.description);
+                    }
+                });
+            }
+            if (schema.references) {
+                schema.references.forEach((reference) => {
+                    if (reference) {
+                        reference.description = marked(reference.description);
+                    }
+                });
+            }
+            return schema;
+        }
+        return '';
+    }
+
+    getInputs(type, inputs) {
+        var results = [];
+        var index = 0;
+
+        inputs.slice(index).forEach((input) => {
+            const name = (index === 0) ? 'input' : ('(' + index.toString() + ')');
+            results.push({
+                name: name,
+                connections: [{id: input}]
+            });
+            index++;
+        });
+
+        return results;
+    }
+
+    getOutputs(type, outputs, layerName) {
+        var results = [];
+        var index = 0;
+
+        outputs.slice(index).forEach((output) => {
+            const name = (index === 0) ? 'output' : ('(' + index.toString() + ')');
+            results.push({
+                name: name,
+                connections: [{id: layerName}]
+            });
+            index++;
+        });
+        return results;
+    }
+
+    getAttributeVisible(operator, name, value) {
```

Comment:
This got moved to the .Attribute classes for most formats as it's only called once, e.g. try to do the JSON reading in the calling code.

Reply:
ok, will check

Reply:
@lutzroeder this is another thing I need you to help with. could you provide an example or explain a bit more what you expect me to do?

Reply:
Maybe check the code for some of the other frameworks like ONNX. The OpenVino code doesn't load any metadata in the 'OperatorMetadata' constructor implementation and won't do anything. For most other frameworks there is an `onnx-metadata.json` file which contains metadata such a the names for inputs (or the weight tensors which currently seems to be not showing in OpenVINO), coloring of operators via groups, documentation strings or which attributes should not show in the graph as they have default values. Maybe open a few `.onnx` files which have colors, show tensors and look more compact as they filter unnecessary attributes. Most of the `-metadata.json` files get updated via scripts under `./tools` when this is automatically possible but some data like the color groups needs to be manually edited.

Reply:
@lutzroeder now I am not using the -metadata.json file, so that this comment is no longer actual, however, I added it to the list of future PRs.

can we close the discussion now?

Reply:
Remove the code maybe until it is being used. Otherwise it will have to be maintained and updated while it is not being used...

Reply:
@lutzroeder I removed the method and the usage.

is the discussion resolved now?